### ISSUE TITLE
Script to scrap LinkedIn Endorsements 

### ISSUE THREAD

**Issue by HarshCasper** [HarshCasper/Rotten-Scripts#1022]
# Description

Create a Script to scrap LinkedIn Endorsements and see who has endorsed you in what technologies. 

## Language

- [X] Golang
- [X] Javascript
- [X] Python
- [X] BASH

**Comment by jhamadhav** [comment]
@HarshCasper can I work on this?

**Comment by rahulraikwar00** [comment]
Any update? @jhamadhav

**Comment by jhamadhav** [comment]
It is in progress, I will take some more time.
I am able to get linkedIn profile page and working on to get the skills and the people that have endorsed it.
It should be done soon

**Comment by rahulraikwar00** [comment]
> It is in progress, I will take some more time.
> I am able to get linkedIn profile page and working on to get the skills and the people that have endorsed it.
> It should be done soon 

If you need any kind of help ..let us know.
But i would suggest 
you can use linkdin API and try to catch profile From there
it would be easy this way
rather than digging and trying to catch every element by own.

**Comment by jhamadhav** [comment]
> 
> 
> > It is in progress, I will take some more time.
> > I am able to get linkedIn profile page and working on to get the skills and the people that have endorsed it.
> > It should be done soon
> 
> If you need any kind of help ..let us know.
> But i would suggest
> you can use linkdin API and try to catch profile From there
> it would be easy this way
> rather than digging and trying to catch every element by own.

wait we can use API 😅 
I thought we can only get basic info with API...
I kinda got blocked for a day too while scrapping 😅 
I will look into the API again and ask in the discord channel if I get stuck 👍

**Comment by rahulraikwar00** [comment]
@jhamadhav  It would be much easier with [this](https://pypi.org/project/linkedin-scraper/)
If you code in python!

**Comment by jhamadhav** [comment]
I will look into it 👍

**Comment by jhamadhav** [comment]
@rahulraikwar00 I tried the python library, it gives the general data but not the endorsement and stuff 😞 
I will retry with the `nodejs` scrapper and ask for help in discord if I get stuck 👍

**Comment by jhamadhav** [comment]
@rahulraikwar00 @HarshCasper 
I am kinda stuck in this project 
I am using puppeteer to scrape a LinkedIn website.
I have succeeded in log in to it and going to the desired profile.

SInce the page is dynamic it loads stuff only after we scroll as I tried this but I'm unable to scroll.
let page = await browser.newPage()
    await page.goto(${url}, { waitUntil: 'networkidle0', timeout: 0 })
    page.scrollBy(0, distance)
What should I do ?

I tried asking for help on the discord but no response :(
